src/maps/savedSearchCreate.ts

```diff
@@ -14,8 +16,49 @@ function mapSavedSearchFiltersToSavedSearchCreate(filters: SavedSearchFilter | u
     return []
   }
 
-  const values = Object.entries(filters)
-  return values.filter(([, value]) => !!value).map(([property, value]) => {
-    return { property, value, object: 'flowRun', type: '', operation: '' }
-  })
+  const values: SavedSearchFilterResponse[] = [
+    getObjectFilter(filters, 'flow'),
+    getObjectFilter(filters, 'tag'),
+    getObjectFilter(filters, 'deployment'),
+    getObjectFilter(filters, 'workPool'),
+    getObjectFilter(filters, 'workQueue'),
+    getObjectFilter(filters, 'workQueue'),
+    getRangeFilter(filters.range),
+  ].filter(isNotNullish)
+
+  return values
+}
+
+type SavedSearchProperty = 'flow' | 'tag' | 'deployment' | 'workQueue' | 'workPool' | 'state'
+
+function getObjectFilter(filters: SavedSearchFilter, property: SavedSearchProperty): SavedSearchFilterResponse | null {
+  const value = filters[property]
+
+  if (!value || isEmptyArray(value)) {
```

Comment:
eslint is complaining about this one, does this need to be
```ts
if (isNullish(value) || isEmptyArray(value))
```
? (i forget if `isNullish` does list comparisons)

Reply:
If eslint is wrong, can you add an ignore?